### ISSUE TITLE
Python SDK Returning None for All Non-ID Value Fields

### ISSUE THREAD

**Issue by dianalingtung** [microsoftgraph/msgraph-sdk-python#945]
### Describe the bug

In the Graph Explorer API I can run the following get request: 

    https://graph.microsoft.com/v1.0/me/memberOf?$search="displayName:{display_name}"&$select=displayName,id
    Headers: ConsistencyLevel: eventual, Prefer: legacySearch=false

Below is my version in the Python SDK:

    request_config = RequestConfiguration(
        options=[ResponseHandlerOption(NativeResponseHandler())]
    )
    request_config.headers.add("ConsistencyLevel", "eventual")
    request_config.headers.add("Prefer", "legacySearch=false")

    url = (f'https://graph.microsoft.com/v1.0/me/'
           f'memberOf?$search="displayName:{displayName}"&$select=displayName,id')

    groups = await (self
                    .user_client
                    .me
                    .with_url(url)
                    .get(request_config))
    groups_data = groups.json()
    groups_value = groups_data['value']
    return groups_value

### Expected behavior

In the Graph API Explorer, I get the following result: 

      {
          "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#directoryObjects(displayName,id)",
          "value": [
              {
                  "@odata.type": "#microsoft.graph.group",
                  "displayName": "{display_name}",
                  "id": "{group_id}"
              }
          ]
      }

In the Python application, I get the following result: 

    [{'@odata.type': '#microsoft.graph.group', 'displayName': None, 'id': '{group_id}'}]

So basically, in the Python code, I am only getting the ID field value. All other field values of interest such as displayName are returned as None. 

Is there some way to alter this code to make the non-ID field values displayable? 

### How to reproduce

Running the example code above should be sufficient to reproduce this scenario. 

### SDK Version

1.11.0

### Debug output

<details><summary>Click to expand log</summary>
```

<Log output here>

```
</details>

**Comment by jrmidkiff** [comment]
Still a problem as of `1.18.0`

**Comment by technaholic** [comment]
Still a problem as of 1.34.0 as well.
Also unable to filter the group query to return only enabled accounts (says filter is not currently supported).
